### ISSUE TITLE
Dependent Job Queueing Order Issue

### ISSUE THREAD

**Issue by jpbakke** [rq/rq#1693]
RQ uses redis SETS store the dependent jobs in the deferred registry.  When the set of dependents is retrieved, the SMEMBER function in redis is used.  Unfortunately, this set function does not guarantee the order of dependents.  I have situation where I want the dependents to be queued in a specific order after their dependency completes and currently this order is non-deterministic due redis using hash storage for sets.  

I would request that the dependent list be stored in redis using ZADD and ZRANGE to retrieve to maintain queuing order for dependents.

**Comment by selwin** [comment]
I'm ok with this change. Do you mind opening a PR for this?

**Comment by jpbakke** [comment]
Yes, I can make the changes and test in a few days and generate a PR for it.

**Comment by selwin** [comment]
Remember to take backwards compatibility into account. If there are dependencies, you'll need to first check the sorted set before falling back to the older set structure.
